### ISSUE TITLE
icp recursive behaviour different across federation

### ISSUE THREAD

**Issue by kript** [irods/irods#4053]
## Bug Report

When using `icp` with the `-r` flag to copy between collections on different Zones, recursion doesn't appear to work the same as within a zone, and gives some odd error messages and misleading exit codes.

I suspect I'm using icp incorrectly, but the behaviour of the `-r` flag and things like trailing '/', wildcards etc is not defined in the help, so its hard to know if I'm just 'holding it wrong'.
However, the fact that there are different messages when trying to copy between zones suggests a different behaviour (and possibly concatting folder names, but that might just be within the error message)

### iRODS Version, OS and Version

Red Hat 7.4, Ubuntu 12.04
4.0.12 (beta), 4.0.10

### What did you try to do?

Recursively copy objects and collections between collections on different zones

### Expected behavior

Recursion works the same way within a zone as across.

### Observed behavior (including steps to reproduce, if applicable)

```
#setup
:~$ imkdir recurse_test2
:~$ icp 1FBLYX_0.json /seq-dev/home/jc18#Sanger1-dev/recurse_test1/
# recursive copy within a Zone
:~$ icp -r /seq-dev/home/jc18#Sanger1-dev/recurse_test1/ /seq-dev/home/jc18#Sanger1-dev/recurse_test2/
:~$ ils /seq-dev/home/jc18#Sanger1-dev/recurse_test2/
/seq-dev/home/jc18#Sanger1-dev/recurse_test2:
  C- /seq-dev/home/jc18#Sanger1-dev/recurse_test2/
#show existing file(s)
:~$ ils /seq-dev/home/jc18#Sanger1-dev/recurse_test1
/seq-dev/home/jc18#Sanger1-dev/recurse_test1:
  1FBLYX_0.json
#create target collection on remote Zone
:~$ imkdir /Sanger1-dev/home/jc18/recurse_test4
#copy recursively with trailing '/'
:~$ icp -r /seq-dev/home/jc18#Sanger1-dev/recurse_test1/ /Sanger1-dev/home/jc18/recurse_test4/
Level 0: collection '/Sanger1-dev/home/jc18/recurse_test4//seq-dev/home/jc18#Sanger1-dev/recurse_test1' is unknown
Level 1: collection '/Sanger1-dev/home/jc18/recurse_test4//seq-dev/home/jc18#Sanger1-dev/recurse_test1' is unknown
#show no files copied
:~$ ils -l /Sanger1-dev/home/jc18/recurse_test4
/Sanger1-dev/home/jc18/recurse_test4:
  C- /Sanger1-dev/home/jc18/recurse_test4/  
#copy recursively *without* trailing '/'
:~$ icp -r /seq-dev/home/jc18#Sanger1-dev/recurse_test1 /Sanger1-dev/home/jc18/recurse_test4
Level 0: collection '/Sanger1-dev/home/jc18/recurse_test4/recurse_test1' is unknown
Level 1: collection '/Sanger1-dev/home/jc18/recurse_test4/recurse_test1' is unknown
#show that copy happened despite errors
:~$ ils -l /Sanger1-dev/home/jc18/recurse_test4/
/Sanger1-dev/home/jc18/recurse_test4:
  C- /Sanger1-dev/home/jc18/recurse_test4/  
  C- /Sanger1-dev/home/jc18/recurse_test4/recurse_test1  
:~$ ils -l /Sanger1-dev/home/jc18/recurse_test4/recurse_test1
/Sanger1-dev/home/jc18/recurse_test4/recurse_test1:
  jc18              0 data          178 2018-08-09.11:25 & 1FBLYX_0.json
# test again and show return code
:~$ imkdir /Sanger1-dev/home/jc18/recurse_test5
2:~$ icp -r /seq-dev/home/jc18#Sanger1-dev/recurse_test1 /Sanger1-dev/home/jc18/recurse_test5
Level 0: collection '/Sanger1-dev/home/jc18/recurse_test5/recurse_test1' is unknown
Level 1: collection '/Sanger1-dev/home/jc18/recurse_test5/recurse_test1' is unknown
:~$ echo $?
0
:~$ imkdir /Sanger1-dev/home/jc18/recurse_test6
:~$ icp -r /seq-dev/home/jc18#Sanger1-dev/recurse_test1/ /Sanger1-dev/home/jc18/recurse_test5/
Level 0: collection '/Sanger1-dev/home/jc18/recurse_test5//seq-dev/home/jc18#Sanger1-dev/recurse_test1' is unknown
Level 1: collection '/Sanger1-dev/home/jc18/recurse_test5//seq-dev/home/jc18#Sanger1-dev/recurse_test1' is unknown
:~$ echo $?
0
```

**Comment by keithj** [comment]
@kript typo in the iRODS version? (`4.1.12`?)

**Comment by trel** [comment]
Same as https://github.com/irods/irods/issues/3547?

**Comment by andrew-irods** [comment]
Addressing this issue depends on issue #3547 being fixed first.

**Comment by alanking** [comment]
This has been resolved along with #4057.
